pkg/operation/seed/seed.go

```diff
@@ -53,6 +53,68 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// NewBuilder returns a new Builder.
+func NewBuilder() *Builder {
+	return &Builder{
+		seedObjectFunc: func() (*gardencorev1beta1.Seed, error) { return nil, fmt.Errorf("seed object is required but not set") },
+	}
+}
+
+// WithSeedObject sets the seedObject attribute at the Builder.
+func (b *Builder) WithSeedObject(seedObject *gardencorev1beta1.Seed) *Builder {
+	b.seedObjectFunc = func() (*gardencorev1beta1.Seed, error) { return seedObject, nil }
+	return b
+}
+
+// WithSeedObjectFromLister sets the seedObject attribute at the Builder after fetching it from the given lister.
+func (b *Builder) WithSeedObjectFromLister(seedLister gardencorelisters.SeedLister, seedName string) *Builder {
```

Comment:
I'm not too sure about how the controller-runtime cache works, maybe you can shed some light on it. I want to prevent that we make unnecessary external calls with the controller-runtime client as we are already watching seeds, etc., and can perfectly read the object from the informers cache. However, as said I'm not sure about the cache of controller-runtime: When does it use it (in contrast to making an actual API call)? We are not watching e.g. seeds using the controller-runtime yet, i.e., do we already have such a cache yet?

Reply:
Sorry, it was a misconception from my side, that the controller-runtime client for the garden cluster uses a cache (it only caches rest clients, but not API objects). That way, we would indeed cause a lot of unnecessary external API calls, which we should prevent.

So, you are probably right, using informers/listers is our best guess for now.
But as soon, as we have cached controller-runtime clients (which we will need anyways, as we are making more and more requests through them), we can probably switch all of these builder methods to controller-runtime clients, to unify and further simplify.

Reply:
Yes, +1, sounds great, thanks.